### ISSUE TITLE
Add Tradacoms EDI support to Ballerina EDI Tool 

### ISSUE THREAD

**Issue by RDPerera** [ballerina-platform/ballerina-library#6082]
**Description:**
Add Tradacoms EDI schema conversion support to the Ballerina EDI Tool.

**Describe your task(s)**
We need to enhance the Ballerina EDI Tool by adding support for Tradacoms EDI schema conversion to Ballerina schema. 

Tradacoms, an early standard for Electronic Data Interchange (EDI), emerged in 1982 in the UK retail sector. Despite ceasing development in 1995, it remains widely used for retail EDI traffic in the UK. It defines 25 transactions, including files for various aspects of retail operations. Although similar to EDIFACT, Tradacoms has distinct syntax and usage, favoring specificity over generality. It employs explicit nesting and lacks qualifiers, tailoring its structure for UK-centric operations. Tradacoms files are specific to the UK market, accommodating sterling currencies and UK tax requirements.

Integrating Tradacoms support into the Ballerina EDI Tool will enable users to seamlessly work with Tradacoms messages within their Ballerina projects.

**Comment by RDPerera** [comment]
### Tradacom EDI
- Tradacoms are primarily used in the UK retail sector.
- It was introduced in 1982.
- One of the precursors of EDIFACT.
- Maintained and extended by the GS1 UK.
- From 1 July 2017 GS1 UK will no longer provide any support for the TRADACOMS.
- Now GS1 only support the  EANCOM, GS1 XML and the GS1 UN/CEFACT XML profile.

### A list of Tradacoms document standards is provided below:

| # | Application Reference | Message type                  |
|---|-----------------------|-------------------------------|
| 1 | ACKHDR                | Acknowledgement               |
| 2 | AVLHDR                | Availability Report           |
| 3 | BTOHDR                | Book Trade Orders             |
| 4 | PVUHDR                | Book Trade Price/Availability Update |
| 5 | CAKHDR                | Claims Acknowledgement        |
| 6 | CLAHDR                | Claims Message                |
| 7 | CORHDR                | Complex Order                 |
| 8 | CREHDR                | Credit Note                   |
| 9 | CREADV                | Credit Advice                 |
| 10| CUSHDR                | Customer Information          |
| 11| DEBADV                | Debit Advice                  |
| 12| DLCHDR                | Delivery Confirmation         |
| 13| DELHDR                | Delivery Notice               |
| 14| DYEHDR                | Dye Instruction               |
| 15| GENHDR                | General Communication         |
| 16| HSOHDR                | Home Shopping Orders          |
| 17| INVFIL                | Invoice                       |
| 18| ISSUES                | Issues                        |
| 19| LPRHDR                | Location Planning Report      |
| 20| PICHDR                | Picking Instruction           |
| 21| ORDHDR                | Purchase Order                |
| 22| PAYORD                | Payment Order                 |
| 23| PRIHDR                | Price Information             |
| 24| PROHDR                | Product Information           |
| 25| PPRHDR                | Product Planning Report       |
| 26| RDAHDR                | Retailer Database             |
| 27| RDBHDR                | Retail B, 1-4 Retailer Database |
| 28| RIFHDR                | Retail Issues File            |
| 29| SADHDR                | Stock Adjustment              |
| 30| SNPSTS                | Stock Snapshot                |
| 31| SRMHDR                | Statement & Remittance Details |
| 32| SORDET                | Supply and Returns Details    |
| 33| SORDAY                | Supply and Returns Summary    |
| 34| SRSHDR                | Supply and Returns Summary    |
| 35| RIFHDR                | Retail Issues File            |
| 36| UCNHDR                | Uplift Confirmation           |
| 37| UPLHDR                | Uplift Instruction            |
| 38| UTLHDR                | Utility Bill                  |

**Comment by RDPerera** [comment]
After contacting GS1, we were able to receive syntax manuals for each of the 25 standards, and since it is a free and openly available standard (even though it has been discontinued), we will publish it [here](https://github.com/RDPerera/tradascom) for future reference. 

However, due to the high number of development hours required (due to the lack of a machine-readable format for the standard) and the lack of entities using it, the development of `Tradacoms EDI support for Ballerina EDI Tool` is de-prioritized. Closing this issue until it is prioritized.
